tests/unit-tests/order/class-wc-tests-orders.php

```diff
@@ -52,4 +52,110 @@ public function test_order_calculate_total_rounding_24695() {
 		$this->assertEquals( 0.79, $order->get_total_tax() );
 	}
 
+	/**
+	 * Test shipping is added and rounded correctly when addedd to total.
+	 *
+	 * @throws WC_Data_Exception When lines cannot be added to order.
+	 */
+	public function test_order_rounding_with_shipping_25748() {
+		update_option( 'woocommerce_prices_include_tax', 'no' );
+		update_option( 'woocommerce_calc_taxes', 'yes' );
+		update_option( 'woocommerce_tax_round_at_subtotal', 'yes' );
+
+		$tax_rate = array(
+			'tax_rate_country'  => '',
+			'tax_rate_state'    => '',
+			'tax_rate'          => '21.0000',
+			'tax_rate_name'     => 'CGST',
+			'tax_rate_priority' => '1',
+			'tax_rate_compound' => '0',
+			'tax_rate_shipping' => '1',
+			'tax_rate_order'    => '1',
+			'tax_rate_class'    => 'tax_1',
```

Comment:
It doesn't work properly because there's no tax class created for it.

Here how to create the tax class:
```php
WC_Tax::create_tax_class( 'tax_1' );
```